src/nvim/os/env.c

```diff
@@ -557,9 +553,6 @@ void home_replace(buf_T *buf, char_u *src, char_u *dst, int dstlen, bool one)
     dirlen = STRLEN(homedir);
 
   homedir_env_orig = homedir_env = (char_u *)os_getenv("HOME");
-  /* Empty is the same as not set. */
-  if (homedir_env != NULL && *homedir_env == NUL)
-    homedir_env = NULL;
```

Comment:
A call to `modify_fname()` could change `homedir_env`. If the pointer is changed, it needs to be freed.

The result for `homedir_env_orig = homedir_env = ""` and `homedir_env_orig = homedir_env = NULL` should be the same. The only difference in the codepath is setting `envlen` to `0` again.